apps/opencs/view/render/scenewidget.hpp

```diff
@@ -3,7 +3,22 @@
 
 #include <map>
 #include <memory>
+#include <mutex>
 
+#ifdef __APPLE__
+#   define __glext_h_
+#   include <QtGui/qopengl.h>
+#   undef __glext_h_
+#   include <QtGui/qopenglext.h>
+#endif
+
+#ifdef WIN32
+//#define __gl_h_
+#include <osg/GL>
+#endif
```

Comment:
This doesn't look right. Both OSG and Qt should be able to hide anything platform-specific like this from you.

Reply:
This part is directly from here https://github.com/openscenegraph/osgQt/blob/2cb70673a4e83a618290e7ee66d52402a94ec3f6/include/osgQOpenGL/osgQOpenGLWidget#L4 but I ripped up that apart and used SceneWidget directly as QOpenGLWidget instead. The idea was to use minimal parts of osgQOpenGLWidget in order to keep things manageable. I'm not sure if this particular part is required, but I decided to keep it because it was already used in osgQt repository.

The thing is that osgQOpenGLWidget can't be used directly, as it uses OSGRenderer, which only supports osgViewer::Viewer and not osgViewer::CompositeViewer. So either OSGRenderer has to be developed to handle both Viewer and CompositeViewer, or  osgQOpenGLWidget has to be able to choose between OSGRenderer and CompositeOsgRenderer (the latter is my modification that uses CompositeViewer).

There are also additional input methods inside osgQOpenGLWidget and OSGRenderer. For starters, I decided to leave all of the input event code out, and use whatever input event methods SceneWidget already relies on. This is also why using osgQOpenGLWidget directly wasn't an option.

Would it perhaps be better to develop osgQOpenGLWidget and use it as member widget of SceneWidget? This is similar to the earlier way how it was done via osgQt GraphicsWindowQT.

Reply:
I'd start by figuring out what's provided by these includes that doesn't come from anywhere else, i.e get rid of them and see what CI yells at you about.